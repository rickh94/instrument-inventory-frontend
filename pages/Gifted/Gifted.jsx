import React, { useState, useLayoutEffect, useContext } from 'react'
import { API } from 'aws-amplify'
import { RootPaper, SearchResultsList, LoadingHeader } from '../../components'
import { HelpersContext } from '../../contexts'

const Gifted = () => {
  const [isLoading, setLoading] = useState(false)
  const [records, setRecords] = useState([])
  const { showAlert } = useContext(HelpersContext)

  useLayoutEffect(() => {
    const getRecords = async () => {
      setLoading(true)
      try {
        const res = await API.get('instrument-inventory', 'filter/gifted')
        setRecords(res)
      } catch (err) {
        showAlert(err)
      }
      setLoading(false)
    }

    getRecords()
  }, [])
  return (
    <RootPaper>
      <LoadingHeader title="Gifted Instruments" isLoading={isLoading} />
      <SearchResultsList results={records} />
    </RootPaper>
  )
}

Gifted.propTypes = {}

export default Gifted
