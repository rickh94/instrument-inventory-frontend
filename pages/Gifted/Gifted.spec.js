import React from 'react'
import Gifted from './Gifted'
import { render, cleanup } from '@testing-library/react'
import { API } from 'aws-amplify'
import { MemoryRouter } from 'react-router-dom'

const flushPromises = new Promise(setImmediate)

const mockInstruments = [
  {
    id: 2,
    number: '4-002',
    type: 'Violin',
    size: '1/4',
    location: 'Storage',
    condition: 4,
    quality: 3,
    gifted: true,
  },
  {
    id: 3,
    number: '3-003',
    type: 'Violin',
    size: '3/4',
    assignedTo: 'Test Name',
    location: 'Office',
    condition: 4,
    quality: 3,
    gifted: true,
    history: ['one', 'two', 'three'],
  },
  {
    id: 4,
    number: '2-004',
    type: 'Violin',
    size: '1/2',
    assignedTo: 'Test Name',
    location: 'Office',
    condition: 4,
    quality: 3,
    gifted: true,
    history: ['one', 'two', 'three'],
  },
  {
    id: 6,
    number: 'C1-002',
    type: 'Cello',
    size: '4/4',
    assignedTo: 'Test Name',
    location: 'Office',
    condition: 4,
    quality: 3,
    gifted: true,
    history: ['one', 'two', 'three'],
  },
  {
    id: 8,
    number: 'C3-004',
    type: 'Cello',
    size: '3/4',
    assignedTo: 'Test Name',
    location: 'Office',
    condition: 4,
    quality: 3,
    gifted: true,
    history: ['one', 'two', 'three'],
  },
]

afterEach(cleanup)

describe('<Gifted />', () => {
  test('renders', () => {
    API.get = jest.fn().mockImplementation(() => Promise.resolve(mockInstruments))
    const { container } = render(
      <MemoryRouter>
        <Gifted showAlert={jest.fn()} />
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })
})
