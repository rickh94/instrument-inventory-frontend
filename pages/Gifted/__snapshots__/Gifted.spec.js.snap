// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gifted /> renders 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded WithStyles(ForwardRef(Paper))-root-1"
  >
    <div
      class="makeStyles-heading-29"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
      >
        Gifted Instruments
      </h5>
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate makeStyles-loader-30"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Instrument
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Size
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Assigned To
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Location
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      />
    </table>
  </div>
</div>
`;
